tests/unit/s2n_shutdown_test.c

```diff
@@ -93,15 +96,19 @@ int main(int argc, char **argv)
 
         /* Verify state prior to alert */
         EXPECT_FALSE(conn->close_notify_received);
-        EXPECT_FALSE(conn->closed);
+        EXPECT_FALSE(s2n_connection_is_closed(conn, S2N_WRITE_CLOSED));
+        EXPECT_FALSE(s2n_connection_is_closed(conn, S2N_READ_CLOSED));
 
         s2n_blocked_status blocked = S2N_NOT_BLOCKED;
         EXPECT_FAILURE_WITH_ERRNO(s2n_shutdown(conn, &blocked), S2N_ERR_IO_BLOCKED);
         EXPECT_EQUAL(blocked, S2N_BLOCKED_ON_READ);
 
         /* Verify state after shutdown attempt */
         EXPECT_FALSE(conn->close_notify_received);
-        EXPECT_TRUE(conn->closed);
+
+        /* Half-close: only write closed */
+        EXPECT_TRUE(s2n_connection_is_closed(conn, S2N_WRITE_CLOSED));
+        EXPECT_FALSE(s2n_connection_is_closed(conn, S2N_READ_CLOSED));
```

Comment:
I think I'm missing something - why does this succeed? Shouldn't `s2n_connection_is_closed(conn, S2N_READ_CLOSED)` return true because the protocol version is < TLS 1.3?

Reply:
We don't actually do a real handshake here, so we never negotiate a version, so it [uses the default](https://github.com/aws/s2n-tls/blob/main/tls/s2n_connection.c#L583), which is TLS1.3.